### PR TITLE
Lab-Anthony: Lab-02 Build Tools

### PR DESCRIPTION
**Pull request by ascasson**
- Added package.json with dependencies/devDependendies
- Added gulpfile
- Added gulp 'default' task to perform mocha/assert testing and linting
- Added gulp 'watch' task
- Replaced eslintrc options to gulp task

### CODE REVIEW

File: lab-anthony/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+.gitignore
+./node_modules
```
Comment (sumtype) [inline lab-anthony/.gitignore]: Good .gitignore, you can also do '/node_modules/**'.

File: lab-anthony/greet-cli.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+'use strict';
+
+const getGreet = require(__dirname + '/lib/greet');
+
+getGreet.greet();
```
Comment (sumtype) [inline lab-anthony/greet-cli.js]: This is fine, in the future you may consider doing this instead for more clarity:
const greet = require(__dirname + '/lib/greet').greet;
That way when debugging it is very clear where that function (greet) comes from.

File: lab-anthony/gulpfile.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const gulp = require('gulp');
+const mocha = require('gulp-mocha');
+const eslint = require('gulp-eslint');
+
+gulp.task('default', function(){
```
Comment (sumtype) [inline lab-anthony/gulpfile.js]: Great job! In the future you may want to break eslint and mocha into separate tasks which can be combined into other tasks (like default).

File: lab-anthony/lib/greet.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+'use strict';
+
+function greet(name){
+  console.log('Hello ' + process.argv[2]);
```
Comment (sumtype) [inline lab-anthony/lib/greet.js]: You should move this console.log statement into the greet-cli.js file to make your function conform exactly to the assignment specifications.

File: lab-anthony/lib/greet.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+  return 'Hello ' + name;
+}
+
+exports.greet = greet;
```
Comment (sumtype) [inline lab-anthony/lib/greet.js]: Great job!  For future consideration, if your module only exports one function it may be simpler to just return that function directly (module.exports = function(...) {...};) so when you require it you don't have to specify .greet to get what you want.

File: lab-anthony/package.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  "description": "using gulp to run mocha/assert/eslint",
+  "main": "greet-cli.js",
+  "directories": {
+    "test": "test"
+  },
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "@ascasson",
+  "license": "ISC",
```
Comment (sumtype) [inline lab-anthony/package.json]: Looks good :), always remember to make sure you're using the license you want to use.

File: lab-anthony/package.json [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  "devDependencies": {
```
Comment (sumtype) [inline lab-anthony/package.json]: You may also want to add 'mocha' itself to the dev dependencies.

File: lab-anthony/test/greet-test.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
+'use strict';
```
Comment (sumtype) [inline lab-anthony/test/greet-test.js]: Looks great, I really like that you tested what happens when the function gets different arguments.  Keep watching your edge cases! :D
